plugin/trino-kafka/src/test/java/io/trino/plugin/kafka/TestKafkaIntegrationSmokeTest.java

```diff
@@ -627,4 +627,35 @@ public String toString()
             return tableName; // for test case label in IDE
         }
     }
+
+    @Test
+    public void testApplyFilterAndLimit()
+    {
+        long partitionNum = (long) computeScalar("SELECT count(distinct _partition_id) partition_num FROM kafka.tpch.customer");
+
+        assertQuery("SELECT _partition_id, _partition_offset" +
```

Comment:
This test is fine to see that there is no regression, but on the other hand there is no prove that limit pushdown actually worked. Can you please use `io.trino.sql.query.QueryAssertions.QueryAssert#isFullyPushedDown` to verify that.